### ISSUE TITLE
Using `buildAllPlatform` more than once in a single node process doesn't reload source files.

### ISSUE THREAD

**Issue by Okazari** [amzn/style-dictionary#404]
Hello there.

I'm currently building a design-token package for my lerna-monorepo that is able to build from a "noBrand" design to a multi brand design. To do so i have a personnal `design-tokens` package. It define the base file for my `design-tokens`. Each of my brand package use my `design-tokens` package as a CLI to build their own design-tokens. They can provide some new file for their brand to surchage some of the token values.

Everything works fine. Until i want to add a `--watch` option to my cli.

To do so i'm using `chokidar` like this :

```js
const path = require('path')
const chokidar = require('chokidar')
const StyleDictionaryPackage = require('style-dictionary')

const baseFilesPath = path.join(__dirname, './src/base/*.json')

const build = ({ buildPath, brandFiles }) => {
  console.log({ buildPath, brandFiles, baseFilesPath })
  const StyleDictionary = StyleDictionaryPackage.extend({
    source: [baseFilesPath, brandFiles],
    platforms: {
      'web/json': {
        transformGroup: 'web',
        buildPath,
        files: [
          {
            destination: 'tokens.json',
            format: 'json/nested',
            filter: (props) => props.attributes.type !== 'global',
          },
        ],
      },
    },
  })

  StyleDictionary.cleanAllPlatforms()
  StyleDictionary.buildAllPlatforms()
}

const watch = (watchPath, args) => {
  chokidar
    .watch(watchPath, { ignoreInitial: true })
    .on('add', () => build(args))
    .on('change', () => build(args))
    .on('unlink', () => build(args))
}

module.exports = (args) => {
  if (args.watch) {
    watch(baseFilesPath, args)
    watch(args.brandFiles, args)
  } else {
    build(args)
  }
}
``` 

Problem is, executing `build()` twice doesn't reflect any changes to any of my watched files.
I investigated a bit and it's due to the way file are loaded.

See this line :
https://github.com/amzn/style-dictionary/blob/f02e505ba77aa7935e558068eaaf958407efafbc/lib/utils/combineJSON.js#L45

I think it's due to some node optimisation. When running twice on a similiar `require()` it apear that it use cache.

I'm able to solve this by replacing `require()` with `JSON.parse(fs.readFileSync(, { encoding: "utf8" }))`

But as you support JS files, i can't decently make a PR for this workaround.

Do you have any idea ? Thanks !

EDIT: 
i found something else.
Adding 
```js
  delete require.cache[resolvedPath]
```
just before the `require` force to load the new version of the file.

I will make a PR accordingly 😄

**Comment by muhammadsamir11** [comment]
I add delete require.cache[resolvedPath] and still the same problem

**Comment by chazzmoney** [comment]
Fixed in #405 

Thank you!

**Comment by dbanksdesign** [comment]
@muhammadsamir11 can you provide some more details, where are you adding that code and what problem are you having?
